src/coreclr/jit/lower.cpp

```diff
@@ -3312,6 +3302,52 @@ GenTree* Lowering::LowerJTrue(GenTreeOp* jtrue)
     }
 #endif
 
+#if defined(TARGET_LOONGARCH64)
+    // for LA64's integer compare and condition-branch instructions,
+    // it's very similar to the IL instructions.
+    if (!varTypeIsFloating(relopOp1->TypeGet()))
+    {
+        relop->SetOper(GT_JCMP);
+        relop->gtType = TYP_VOID;
+
+        relop->gtFlags &= ~(GTF_JCMP_TST | GTF_JCMP_EQ | GTF_JCMP_MASK);
+        relop->gtFlags |= (GenTreeFlags)(cond.GetCode() << 25);
+
+        if ((cond.GetCode() == GenCondition::EQ) || (cond.GetCode() == GenCondition::NE))
+        {
+            relop->gtFlags &= ~GTF_UNSIGNED;
+        }
+
+        if (relopOp2->IsCnsIntOrI())
+        {
+            relopOp2->SetContained();
+        }
+
+        BlockRange().Remove(jtrue);
+
+        assert(relop->gtNext == nullptr);
+        return nullptr;
+    }
+    else
+    {
+        // But the LA64's float compare and condition-branch instructions,
+        // it has the condition flags indicating the comparing results.
+        relop->gtType = TYP_VOID;
+        relop->gtFlags |= GTF_SET_FLAGS;
+
+        if (relop->OperIs(GT_TEST_EQ))
+        {
+            relop->SetOper(GT_EQ);
+        }
+        else if (relop->OperIs(GT_TEST_NE))
+        {
+            relop->SetOper(GT_NE);
+        }
+        assert(relop->OperIs(GT_EQ, GT_NE, GT_LT, GT_LE, GT_GE, GT_GT));
```

Comment:
You should never see `GT_TEST_EQ` and `GT_TEST_NE` on LA64, the only place that creates these nodes is `OptimizeConstCompare` and it only does it for `TARGET_XARCH || TARGET_ARM64`. Also, the semantics of this node is `(a & b) ==/!= 0`, so changing it to `GT_EQ/GT_NE` would not make sense.

```suggestion
        // But the LA64's float compare and condition-branch instructions,
        // it has the condition flags indicating the comparing results.
        relop->gtType = TYP_VOID;
        relop->gtFlags |= GTF_SET_FLAGS;
        assert(relop->OperIs(GT_EQ, GT_NE, GT_LT, GT_LE, GT_GE, GT_GT));
```

Reply:
Thanks

Reply:
I also delete GT_TEST* nodes within the codegenloongarch64.cpp